subsys/net/ip/net_context.c

```diff
@@ -689,6 +689,49 @@ int net_context_put(struct net_context *context)
 	return 0;
 }
 
+static int bind_default(struct net_context *context,
+				    const struct sockaddr *dest_addr)
+{
+	struct net_if *iface = net_if_get_default();
```

Comment:
> If we have multiple interfaces, then this selection might get wrong interface.

Yes, it might. Just the same as net_context_bind(). My assumption is that if explicit bind() is not called, it's called implicitly as bind(INADDR_ANY, rand()). bind_default() just performs the same operations as next_context_bind(INADDR_ANY, rand()), using the same code (the reason I factored out code into new function instead of just calling next_context_bind() because it seemed like a chore to construct INADDR_ANY for IPv4 vs IPv6 cases; maybe that was wrong actually).

I can paste your code in, but the only expected outcome from that would be that implementation (and thus behavior) of explicit vs implicit bind diverge, and thus lead to confusion. Note that neither next_context_bind() nor your proposed change to bind_default() is correct impl of INADDR_ANY - per POSIX, it should bind to *all* not to default/"best" interface.